frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/portlets/AttachmentAwarePortlet.java

```diff
@@ -95,27 +109,63 @@ protected void dealWithAttachments(ResourceRequest request, ResourceResponse res
     }
 
     private void doGetAttachmentForDisplay(ResourceRequest request, ResourceResponse response) throws IOException, PortletException {
-        String attachmentId = request.getParameter(PortalConstants.ATTACHMENT_ID);
+        final String attachmentId = request.getParameter(PortalConstants.ATTACHMENT_ID);
         final User user = UserCacheHolder.getUserFromRequest(request);
+        final TSerializer serializer = new TSerializer(new TSimpleJSONProtocol.Factory());
 
         Attachment attachment = attachmentPortletUtils.getAttachmentForDisplay(user, attachmentId);
 
         if(attachment==null) {
             response.setProperty(ResourceResponse.HTTP_STATUS_CODE, "500");
         } else {
-            request.setAttribute(PortalConstants.DOCUMENT_TYPE, getDocumentType(request));
-            request.setAttribute(PortalConstants.ATTACHMENTS, toSingletonSet(attachment));
-            include("/html/utils/ajax/attachmentsAjax.jsp", request, response, PortletRequest.RESOURCE_PHASE);
+            try {
+                writeJSON(request, response, serializer.toString(attachment));
+            } catch (TException exception) {
+                LOGGER.error("Cannot serialize attachment", exception);
+                response.setProperty(ResourceResponse.HTTP_STATUS_CODE, "500");
+            }
         }
     }
 
     private void serveAttachmentSet(ResourceRequest request, ResourceResponse response) throws IOException, PortletException {
-        String documentId = request.getParameter(PortalConstants.DOCUMENT_ID);
+        final String documentType = getDocumentType(request);
+        final String documentId = request.getParameter(PortalConstants.DOCUMENT_ID);
         final User user = UserCacheHolder.getUserFromRequest(request);
+        final TSerializer serializer = new TSerializer(new TSimpleJSONProtocol.Factory());
+
+        // this is the raw attachment data
+        List<String> attachments = getAttachments(documentId, documentType, user).stream().sorted((left, right) -> {
+            return left.getFilename().compareTo(right.getFilename());
+        }).map(attachment -> {
+            try {
+                return serializer.toString(attachment);
+            } catch (TException exception) {
+                return "{ \"error\": \"" + TagUtils.escapeAttributeValue(exception.toString()) + "\" }";
+            }
+        }).collect(Collectors.toList());
+
+        // enum data to be able to interpret attachment type
+        JSONObject attachmentTypes = JSONFactoryUtil.createJSONObject();
```

Comment:
Is it possible that Jackson is not a dependency of the portlet project yet? In this case I need to add it. Is this ok?